doc/source/ray-core/fault_tolerance/objects.rst

```diff
@@ -0,0 +1,35 @@
+Objects fault tolerance
+=======================
+
+There are two components involved for an object, the data and the metadata.
+The data is stored in the object store and the metadata is stored in the
+:ref:`owner <fault-tolerance_ownership>` of the object.
+
+Lost in the object store
+------------------------
+
+When the object is lost in the object store, Ray will use
+*lineage reconstruction* to recover the object. This can occur during node
+failures. Ray will first automatically attempt to recover the value by looking
+for copies of the same object on other nodes. If none are found, then Ray will
+automatically recover the value by re-executing the task that created the value.
+Arguments to the task are recursively reconstructed with the same method. What's
+worth noting here is that Ray is not able to recover the objects not generated
+by tasks, for example, the object which is generated by ``ray.put``.
+
```

Comment:
We need to elaborate the scenario here a bit more. As written is unclear how that may arise. 

Since the task explicitly did not create an object X, but some other "entity" did a `ray.put(X)`, what other entity that might be? And why that "entity" cannot recover it if it created it?